src/test/java/com/google/api/codegen/testsrc/common/library.proto

```diff
@@ -674,15 +659,15 @@ message GetBookRequest {
   // The name of the book to retrieve.
   string name = 1 [
     (google.api.field_behavior) = REQUIRED,
-    (google.api.resource_reference) = "Book"];
+    (google.api.resource_reference).type = "library.googleapis.com/Book"];
```

Comment:
How are we differentiating between fields that are of the "resource set" variety and those that are just a single resource? 

Whether a field is a resource set or a singular resource mighttt change the field name in C# and be a breaking change. The latest master branch has the C# baseline file that we could see if there's a diff in.

Reply:
Fields with a single pattern are single, with multiple patterns are sets. This behaviour can also be modified using the History field in ResourceDescriptor.

Reply:
Does `library.googleapis.com/Book` refer to the resource set? This field `name` should only refer to a single resource.

Reply:
It is not possible to refer to only a single resource from a set under the new annotation. To get that behavior, would need to use a field_name_pattern in the v2 yaml

Reply:
How would a single resource be referred to by the v2 yaml?